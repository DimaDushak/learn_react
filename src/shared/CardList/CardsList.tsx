import React from 'react';
import { usePostsData } from '../../utils/hooks';
import Card from './Card';
import styles from './cardslist.css';

export default function CardsList(): JSX.Element {
    const postsData = usePostsData();

    return (
        <ul className={styles.cardsList}>
            {postsData.map(({ data }) => (
                <Card key={data.id} post={data} />
            ))}
        </ul>
    );
}
