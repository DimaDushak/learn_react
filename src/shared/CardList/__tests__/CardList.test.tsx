import { render } from '@testing-library/react';
import React from 'react';
import CardsList from '../CardsList';

test('should render CardsList (snapshot)', () => {
    const { container } = render(<CardsList />);

    expect(container.firstChild).toMatchSnapshot();
});
