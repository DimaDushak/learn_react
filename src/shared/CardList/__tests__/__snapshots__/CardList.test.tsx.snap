// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CardsList (snapshot) 1`] = `
<ul
  class="cardsList"
/>
`;
